opensrp-anc/src/main/java/org/smartregister/anc/library/activity/MainContactActivity.java

```diff
@@ -90,6 +93,12 @@ private void initializeMainContactContainers() {
             process(contactForms);
             requiredFieldsMap.put(ConstantsUtils.JsonFormUtils.ANC_TEST_TASKS_ENCOUNTER_TYPE, 0);
 
+            if (StringUtils.isNotBlank(formInvalidFields) && contactNo > 1 && !PatientRepository.isFirstVisit(baseEntityId)) {
+                String[] pair = formInvalidFields.split(":");
```

Comment:
Should check the size of this string array. Otherwise, you will get an out of bounds exception